components/constellation/constellation.rs

```diff
@@ -204,6 +204,20 @@ pub struct Constellation<Message, LTF, STF> {
     /// The last frame tree sent to WebRender.
     active_browser_id: Option<TopLevelBrowsingContextId>,
 
+    /// All the browsers in constellation. Since the active browser
+    /// (`active_browser_id`) may change with the `SelectBrowser` message (e.g.
+    /// when user switches to another tab), we must keep track of each browser's
+    /// focused browsing context so that we can restore the
+    /// `focused_browsing_context_id` to point to the active browser's focused
+    /// browsing context.
+    browser_ids: HashMap<TopLevelBrowsingContextId, BrowsingContextId>,
```

Comment:
It would be good to change the name here to something more along the lines of `focused_browsing_contexts`